src/mut/morphology.cpp

```diff
@@ -255,6 +255,23 @@ void Morphology::sanitize(const morphio::readers::DebugInfo& debugInfo) {
     }
 }
 
+void Morphology::_raiseIfUnifurcations() {
+    for (auto it = depth_begin(); it != depth_end(); ++it) {
+        std::shared_ptr<Section> section_ = *it;
+        if (section_->isRoot())
+            continue;
+
+        unsigned int parentId = section_->parent()->id();
+
+        auto parent = section_->parent();
+        bool isUnifurcation = parent->children().size() == 1;
+
+        if (isUnifurcation) {
+            throw WriterError(readers::ErrorMessages().ERROR_ONLY_CHILD_SWC_WRITER(parentId));
+        }
+    }
+}
+
```

Comment:
So this could be:
```suggestion
bool Morphology::hasUnifurcations() {
    for (auto it = depth_begin(); it != depth_end(); ++it) {
        std::shared_ptr<Section> section_ = *it;
        if (section_->isRoot())
            continue;

        unsigned int parentId = section_->parent()->id();

        auto parent = section_->parent();
        if (parent->children().size() == 1) return true;    
    }
    return false;
}
void Morphology::_raiseIfUnifurcations() {
    if (hasUnifurcations()) {
        throw WriterError(readers::ErrorMessages().ERROR_ONLY_CHILD_SWC_WRITER(parentId));
    }
}
```

Reply:
+1 for this indeed

Reply:
agreed; can you add that to the review?